awscli/clidriver.py

```diff
@@ -465,29 +476,31 @@ def _show_error(self, msg):
 
     def _handle_top_level_args(self, args):
         emit_top_level_args_parsed_event(self.session, args)
-        if args.profile:
+        if getattr(args, 'profile', False):
             self.session.set_config_variable('profile', args.profile)
-        if args.region:
+        if getattr(args, 'region', False):
             self.session.set_config_variable('region', args.region)
-        if args.debug:
-            # TODO:
-            # Unfortunately, by setting debug mode here, we miss out
-            # on all of the debug events prior to this such as the
-            # loading of plugins, etc.
-            self.session.set_stream_logger('botocore', logging.DEBUG,
-                                           format_string=LOG_FORMAT)
-            self.session.set_stream_logger('awscli', logging.DEBUG,
-                                           format_string=LOG_FORMAT)
-            self.session.set_stream_logger('s3transfer', logging.DEBUG,
-                                           format_string=LOG_FORMAT)
-            self.session.set_stream_logger('urllib3', logging.DEBUG,
-                                           format_string=LOG_FORMAT)
+        self._set_logging(getattr(args, 'debug', False))
+
+    def _set_logging(self, debug):
+        if debug:
+            set_stream_logger('botocore', logging.DEBUG,
+                              format_string=LOG_FORMAT)
+            set_stream_logger('awscli', logging.DEBUG,
+                              format_string=LOG_FORMAT)
+            set_stream_logger('s3transfer', logging.DEBUG,
+                              format_string=LOG_FORMAT)
+            set_stream_logger('urllib3', logging.DEBUG,
+                              format_string=LOG_FORMAT)
             LOG.debug("CLI version: %s", self.session.user_agent())
             LOG.debug("Arguments entered to CLI: %s", sys.argv[1:])
-
         else:
-            self.session.set_stream_logger(logger_name='awscli',
-                                           log_level=logging.ERROR)
+            remove_stream_logger('botocore')
```

Comment:
To make this a bit easier to manage, let's:
1) Keep a list of packages we need to add handlers for and programmatically add and remove handlers using this list
2) Add a comment in the code on why we need to remove the stream handlers. It definitely was not obvious why it was needed.